src/DotLiquid.Tests/HashTests.cs

```diff
@@ -117,5 +118,50 @@ public void TestUpperTwoScenarioWithSameClass()
             IncludeBaseClassPropertiesOrNot(true);
         }
         #endregion
+
+        [Test]
+        public void TestDefaultConstructor()
+        {
+            var hash = new Hash(0); // default value of zero
+            hash["key"] = "value";
+
+            Assert.False(hash.Contains("unknown-key"));
+            Assert.False(hash.ContainsKey("unknown-key"));
```

Comment:
For clarification - this is just temporarily false but behavior change will be introduced in #441

Reply:
Yes, I had to change from true to false whilst reverting perf changes.

Reply:
Please add a code comment `//` above so future maintainers know if/that it is ok to change.

Reply:
Comment added.